lib/models/api_calls.dart

```diff
@@ -80,20 +83,20 @@ class ApiCalls {
   ///
   /// This method returns a [DetailedDevice] object.
   dynamic getDetailedDevice(String authToken, String deviceId) async {
+    var response;
     authToken = 'Bearer $authToken';
-    try {
-      final response = await _client.get(DETAILED_DEVICE_URL + deviceId,
-          headers: {HttpHeaders.authorizationHeader: authToken});
-      // If conection is succesful return a DetailedDevice object by parsing the
-      // response body
-      if (response.statusCode == 200) {
-        return DetailedDevice.fromMap(json.decode(response.body));
-      } else {
-        // If that call was not successful, throw an error.
-        throw Exception(response.statusCode);
-      }
-    } finally {
-      _client.close();
+    _client == null
```

Comment:
Also, why can the client be null? Shouldn't it always be provided? And if just calling http get works, why do we need the client then?

Reply:
When calling http.get it auto generates a http.Client and after the call it closes the client so it don't get hanged. The only moment that I need to provide a client is in testing since I need to mock the calls.

Reply:
I see, and it's not possible to mock http in the tests?